eks/extra-variables.tf

```
@@ -1,8 +1,18 @@
 variable "logs_storage" {
-  description = "AWS S3 bucket configuration values for the bucket where the logs will be stored."
+  description = <<-EOT
+    AWS S3 bucket configuration values for the bucket where the logs will be stored.
+
+    An IAM role is required to give the Loki components read and write access to the S3 bucket. You can create this role yourself or let the module create it for you. If you want the module to create the role, you need to provide the OIDC issuer's URL for the EKS cluster. If you create the role yourself, you need to provide the ARN of the IAM role you created.
+  EOT
   type = object({
-    bucket_id    = string
-    region       = string
-    iam_role_arn = string
+    bucket_id               = string
+    create_role             = bool
```

```suggestion
    create_role             = optional(bool, false)
```
This would make the change backwards compatible, as far as I can tell.

---

It would be a breaking change regardless. I think the addition of that attribute would not be a problem but the fact that I removed the `region` attribute.

---

Indeed, I didn't see that. So my suggestion doesn't make much of a difference.

---

Thank you, @ckaenzig!